element-connector/src/main/java/org/eclipse/californium/elements/util/Bytes.java

```diff
@@ -105,7 +114,10 @@ public final boolean equals(Object obj) {
 	 * @return bytes as (hexadecimal) string
 	 */
 	public final String getAsString() {
-		return StringUtil.byteArray2Hex(bytes);
+		if (asString == null) {
+			asString = StringUtil.byteArray2Hex(bytes);
+		}
+		return asString;
```

Comment:
Remove it and run benchmarks.

Reply:
This is not a real cooperative answer. Anyway ...

Reply:
Why?

Run benchmarks and give us your feedback, if you get a performance gain or not.
If you get a gain, let us know, if you feel that it is too small.
You may also check the memory usage, if this is affected by that.

And it's even easy, to remove that later, if someone has spend the time into check for benchmarks and heap-consumption.

"cooperative" seems for me to include, that not only the other two are working.

Reply:
At first sight I don't see the link between this optimization and the PR itself.
I would be surprised it could have significant benefits and so I suspect this could maybe be a premature optimization. (anybody could fall in this trap)

So when I ask, this I expect several kind of answer : 
 - I don't know, I didn't test it. (this smell premature optimization)
 - I have some hint that could help but I need to test it more. (Ok all is in control)
 - I test it and my result are significant. (ok all is in control)

But not really : "test your self" without any other explanation.
Imagine the reverse situation, I create a PR with something that you could find as potential premature optimization and instead of explaining why I do that I just tell you : "test yourself"

You would be comfortable with this ?

Reply:
No, I always run benchmarks and I got a small penalty using this or the other PR (45.000 down to 41.000). But there is also a other "time eating" candidate, it's the MDC, itself if multiple entries are added.
To check that, takes time. And therefore I postponed the further anlaysis of that.